### ISSUE TITLE
Support ARM for Hi3798 V200+

### ISSUE THREAD

**Issue by yangfl** [histb-mainline/linux#5]
Hi3798 V200+ come from Cortex-A9. Of course AArch64 is preferred, but hisi sdk also comes from ARM32 support for those chips.

Should we add those chips for ARM32?

**Comment by 185264646** [comment]
> Hi3798 V200+ come from Cortex-A9. Of course AArch64 is preferred, but hisi sdk also comes from ARM32 support for those chips.
> 
> Should we add those chips for ARM32?

yes, but it's not in high priority. for those boards with low memory this might make sense.

**Comment by 185264646** [comment]
> Hi3798 V200+ come from Cortex-A9. Of course AArch64 is preferred, but hisi sdk also comes from ARM32 support for those chips.
> 
> Should we add those chips for ARM32?

不太容易。阻力主要来自U-Boot。U-Boot的维护者不想支持64位U-Boot引导32位内核，反过来也是。搞不好要自己维护boot loader才有可能了。

支持ARM32是有一定好处的，因为ARM64过了ATF之后就变成non-secure状态了，据我测试non-secure状态下有很多硬件访问不了。寄存器读出来全是0.
